src/gui/qgsvertexmarker.cpp

```diff
@@ -58,7 +59,8 @@ void QgsVertexMarker::setPenWidth( int width )
 
 void QgsVertexMarker::paint( QPainter *p )
 {
-  qreal s = ( mIconSize - 1 ) / 2.0;
+  int scaledIconSize = QgsGuiUtils::scaleIconSize( mIconSize );
+  qreal s = ( scaledIconSize - 1 ) / 2.0;
```

Comment:
I am wondering if this should not be considered as API break. In some plugins I have been expecting that the icon size is still in pixels, so the plugin code does scaling based on DPI by itself for vertex markers and rubber bands...

This change should be probably consistent with QgsRubberBand, so both use dpi "independent" pixel sizes / widths. Maybe best to defer to a separate PR?

Reply:
@wonder-sk , OK, I'll do a separate PR.